catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavEntry.java

```diff
@@ -0,0 +1,324 @@
+
+    private final DavEntry parent;
+
+    private DavEntry[] children;
+
+    private File file;
+
+    private String location;
+
+    private boolean exists;
+
+    private boolean directory;
+
+    private long lastModified;
+
+    private long length;
+
+    private String eTag;
+
+    /**
+     * @param location must be fully qualified
+     */
+    DavEntry(final String location) {
+        this(null, location);
+    }
+
+    private DavEntry(final DavEntry parent, final String location) {
+        if (location == null) {
+            throw new IllegalArgumentException("File is missing");
+        }
+        this.parent = parent;
+        this.setLocation(getLocation(location, parent));
+    }
+
+    static DavEntry[] getEmptyEntries() {
+        return EMPTY_ENTRIES;
+    }
+
+    boolean refresh(DavResource davResource) {
+
+        // cache original values
+        final boolean origExists = isExists();
+        final long origLastModified = getLastModified();
+        final boolean origDirectory = isDirectory();
+        final long origLength = getLength();
+        final String origEtag = getETag();
+
+        // refresh the values
+        setExists(davResource != null);
+        setDirectory(isExists() && davResource.isDirectory());
+        setLastModified(isExists() && davResource.getModified() != null ?
+                davResource.getModified()
+                        .getTime() :
+                0);
+        setLength(isExists() && !isDirectory() ? davResource.getContentLength() : 0);
+        setETag(isExists() ? davResource.getEtag() : "0");
+
+        if (file != null && !isDirectory() && origLastModified != getLastModified()) {
+            FileUtils.deleteQuietly(file.getParentFile());
+        }
+
+        // Return if there are changes
+        return isExists() != origExists //
+                || getLastModified() != origLastModified //
+                || isDirectory() != origDirectory //
+                || getLength() != origLength //
+                || !Objects.equals(getETag(), origEtag);
+    }
+
+    DavEntry newChildInstance(String location) {
+        return new DavEntry(this, location);
+    }
+
+    // construct the fully qualified location from a fully qualified parent and
+    // a child relative to the parent
+    static String getLocation(String initialLocation, DavEntry parent) {
+        String location = initialLocation;
+        if (parent != null && !location.startsWith("http")) {
+            String parentLocation = parent.getLocation();
+            if (parentLocation.endsWith("/") && location.startsWith("/")) {
+                location = location.replaceFirst("/", "");
+            }
+            if (!parentLocation.endsWith("/") && !location.startsWith("/")) {
+                location = "/" + location;
+            }
+            location = parentLocation + location;
+        }
+        try {
+            URL url = new URL(location);
+            URI uri = new URI(url.getProtocol(),
```

Comment:
Yes, the internal logic has different behaviors for handling character encoding/decoding. I will try to add an explanatory comment.